js/util/pipeline-fixed.js

```diff
@@ -0,0 +1,378 @@
+ *
+ * - `INTEREST_TIMEOUT`: if any of the Interests times out (probably after a number of retries)
+ * - `DATA_HAS_NO_SEGMENT`: if any of the retrieved Data packets don't have a segment
+ *   as the last component of the name (not counting the implicit digest)
+ * - `SEGMENT_VERIFICATION_FAILED`: if any retrieved segment fails
+ *   the user-provided VerifySegment callback or KeyChain verifyData.
+ * - `IO_ERROR`: for I/O errors when sending an Interest.
+ * -`DUPLICATE_SEGMENT_RECEIVED`: if any duplicate segment is receveid during
+ *   fetching the data (we do not ask for a segment that is already retrieved,
+ *   again)
+ *
+ * In order to validate individual segments, a KeyChain needs to be supplied.
+ * If verifyData fails, the fetching process is aborted with
+ * SEGMENT_VERIFICATION_FAILED. If data validation is not required, pass null.
+ *
+ *
+ * This is a public constructor to create a new PipelineFixed.
+ * If validatorKeyChain is not null, use it and ignore verifySegment.
+ * @param {string} basePrefix This is the prefix of the data we want to retreive
+ * its segments (excluding <version> and <segment> components).
+ * @param {Face} face The segments will be fetched through this face.
+ * @param validatorKeyChain {KeyChain} If this is not null, use its verifyData
+ * instead of the verifySegment callback.
+ * @param {function} verifySegment When a Data packet is received this calls
+ * verifySegment(data) where data is a Data object. If it returns False then
+ * abort fetching and call onError with
+ * PipelineFixed.ErrorCode.SEGMENT_VERIFICATION_FAILED.
+ * NOTE: The library will log any exceptions thrown by this callback, but for
+ * better error handling the callback should catch and properly handle any
+ * exceptions.
+ * @param {function} onComplete When all segments are received, call
+ * onComplete(content) where content is a Blob which has the concatenation of
+ * the content of all the segments.
+ * NOTE: The library will log any exceptions thrown by this callback, but for
+ * better error handling the callback should catch and properly handle any
+ * exceptions.
+ * @param {function} onError Call onError.onError(errorCode, message) for
+ * timeout or an error processing segments. errorCode is a value from
+ * PipelieFixed.ErrorCode and message is a related string.
+ * NOTE: The library will log any exceptions thrown by this callback, but for
+ * better error handling the callback should catch and properly handle any
+ * exceptions.
+ * @constructor
+ */
+var PipelineFixed = function PipelineFixed
+  (basePrefix, face, validatorKeyChain, verifySegment, onComplete, onError)
+{
+  this.contentPrefix = basePrefix;
+  this.face = face;
+  this.validatorKeyChain = validatorKeyChain;
+  this.verifySegment = verifySegment;
+  this.onComplete = onComplete;
+  this.onError = onError;
+
+  this.contentParts = []; // of Buffer
+
+  this.satisfiedSegments = new Set(); // no duplicate entry
```

Comment:
`Set` type is ES6 only. According to [this answer](https://www.lists.cs.ucla.edu/pipermail/ndn-lib/2018-October/000452.html), ndn-js must work in ES5. Therefore, you need to load a polyfill.

Reply:
Changed the implementation to `array` to be supported by ES5 and ES6